### PR TITLE
Fix DelayedLoadUnloadWrapper loaded count being incorrect

### PR DESCRIPTION
**Pull request by peppy**
`DelayedLoadUnloadWrapper` with no containing `OpitimisingContainer` would have incorrectly incremented the counter.

### CODE REVIEW

File: osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs [hunk 1]
```diff
@@ -65,8 +65,12 @@ protected override void EndDelayedLoad(Drawable content)
 
             Debug.Assert(unloadSchedule == null);
 
-            unloadSchedule = OptimisingContainer?.ScheduleCheckAction(checkForUnload);
-            loaded_count.Value++;
+            if (OptimisingContainer != null)
+            {
+                unloadSchedule = OptimisingContainer.ScheduleCheckAction(checkForUnload);
+                Debug.Assert(unloadSchedule != null);
+                loaded_count.Value++;
```
Comment (smoogipoo) [inline osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs]: This seems weird to me, especially considering it's still "DLUWs loaded". How about keeping a separate state for this, something like...

```yaml
bool contentLoaded

void EndDelayedLoad():
    contentLoaded = true;
    loaded_count.Value++;

void CancelTasks():
    if (contentLoaded)
        loaded_count.Value--;
        contentLoaded = false;
```

Comment (peppy) [reply osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs]: I'd rather add a second counter for DelayedLoadWrappers which counts that case.

I actually started this PR as expoding optimised and non-optimised counts, can go down that path as well if you'd prefer.

Or just renaming the variable. This one is specifically about tracking cases where they *can* be unloaded.

Comment (smoogipoo) [reply osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs]: I was about to suggest both of those as alternates. I don't mind either of those but would prefer optimised and non-optimised counts.

Comment (peppy) [reply osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs]: so a count for unoptimised DLUW specifically?

Comment (smoogipoo) [reply osu.Framework/Graphics/Containers/DelayedLoadUnloadWrapper.cs]: Yeah. Would also require renaming this variable ofc.

**Pull request merged**